File: src/main/java/tn/esprit/Pidev3A49/models/Event.java
package tn.esprit.Pidev3A49.models;

import java.time.LocalDate;
import java.time.LocalDateTime;

public class Event {

    private int idEvent;
    private String titre;
    private String description;
    private LocalDate dateEvent;
    private String lieu;
    private int capacite;
    private String typeEvent;
    private String imageEvent;
    private double prixEvent;
    private LocalDateTime createdAt;
    private boolean premium;

    public Event() {
    }

    public Event(String titre, String description, LocalDate dateEvent, String lieu, int capacite,
                 String typeEvent, String imageEvent, double prixEvent, boolean premium) {
        this.titre = titre;
        this.description = description;
        this.dateEvent = dateEvent;
        this.lieu = lieu;
        this.capacite = capacite;
        this.typeEvent = typeEvent;
        this.imageEvent = imageEvent;
        this.prixEvent = prixEvent;
        this.createdAt = LocalDateTime.now();
        this.premium = premium;
    }

    public Event(String titre, String description, LocalDate dateEvent, String lieu, int capacite,
                 String typeEvent, String imageEvent, double prixEvent, LocalDateTime createdAt, boolean premium) {
        this.titre = titre;
        this.description = description;
        this.dateEvent = dateEvent;
        this.lieu = lieu;
        this.capacite = capacite;
        this.typeEvent = typeEvent;
        this.imageEvent = imageEvent;
        this.prixEvent = prixEvent;
        this.createdAt = createdAt;
        this.premium = premium;
    }

    public Event(int idEvent, String titre, String description, LocalDate dateEvent, String lieu, int capacite,
                 String typeEvent, String imageEvent, double prixEvent, LocalDateTime createdAt, boolean premium) {
        this.idEvent = idEvent;
        this.titre = titre;
        this.description = description;
        this.dateEvent = dateEvent;
        this.lieu = lieu;
        this.capacite = capacite;
        this.typeEvent = typeEvent;
        this.imageEvent = imageEvent;
        this.prixEvent = prixEvent;
        this.createdAt = createdAt;
        this.premium = premium;
    }

    public int getIdEvent() {
        return idEvent;
    }

    public void setIdEvent(int idEvent) {
        this.idEvent = idEvent;
    }

    public String getTitre() {
        return titre;
    }

    public void setTitre(String titre) {
        this.titre = titre;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public LocalDate getDateEvent() {
        return dateEvent;
    }

    public void setDateEvent(LocalDate dateEvent) {
        this.dateEvent = dateEvent;
    }

    public String getLieu() {
        return lieu;
    }

    public void setLieu(String lieu) {
        this.lieu = lieu;
    }

    public int getCapacite() {
        return capacite;
    }

    public void setCapacite(int capacite) {
        this.capacite = capacite;
    }

    public String getTypeEvent() {
        return typeEvent;
    }

    public void setTypeEvent(String typeEvent) {
        this.typeEvent = typeEvent;
    }

    public String getImageEvent() {
        return imageEvent;
    }

    public void setImageEvent(String imageEvent) {
        this.imageEvent = imageEvent;
    }

    public double getPrixEvent() {
        return prixEvent;
    }

    public void setPrixEvent(double prixEvent) {
        this.prixEvent = prixEvent;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public boolean isPremium() {
        return premium;
    }

    public void setPremium(boolean premium) {
        this.premium = premium;
    }

    @Override
    public String toString() {
        return "Event{" +
                "idEvent=" + idEvent +
                ", titre='" + titre + '\'' +
                ", dateEvent=" + dateEvent +
                ", lieu='" + lieu + '\'' +
                ", capacite=" + capacite +
                ", typeEvent='" + typeEvent + '\'' +
                ", prixEvent=" + prixEvent +
                ", createdAt=" + createdAt +
                ", premium=" + premium +
                '}';
    }
}

File: src/Entity/Event.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity]
#[ORM\Table(name: 'events')]
class Event
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(name: 'id_event')]
    private ?int $id = null;

    #[ORM\Column(length: 150)]
    private string $titre;

    #[ORM\Column(type: 'text')]
    private string $description;

    #[ORM\Column(name: 'date_event', type: 'date')]
    private \DateTimeInterface $dateEvent;

    #[ORM\Column(length: 150)]
    private string $lieu;

    #[ORM\Column]
    private int $capacite;

    #[ORM\Column(name: 'type_event', length: 100)]
    private string $typeEvent;

    #[ORM\Column(name: 'image_event', length: 255, nullable: true)]
    private ?string $imageEvent = null;

    #[ORM\Column(name: 'prix_event', type: 'decimal', precision: 10, scale: 2)]
    private ?string $prixEvent = null;

    #[ORM\Column(name: 'created_at', type: 'datetime_immutable')]
    private \DateTimeImmutable $createdAt;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getTitre(): string
    {
        return $this->titre;
    }

    public function setTitre(string $titre): self
    {
        $this->titre = $titre;
        return $this;
    }

    public function getDescription(): string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;
        return $this;
    }

    public function getDateEvent(): \DateTimeInterface
    {
        return $this->dateEvent;
    }

    public function setDateEvent(\DateTimeInterface $dateEvent): self
    {
        $this->dateEvent = $dateEvent;
        return $this;
    }

    public function getLieu(): string
    {
        return $this->lieu;
    }

    public function setLieu(string $lieu): self
    {
        $this->lieu = $lieu;
        return $this;
    }

    public function getCapacite(): int
    {
        return $this->capacite;
    }

    public function setCapacite(int $capacite): self
    {
        $this->capacite = $capacite;
        return $this;
    }

    public function getTypeEvent(): string
    {
        return $this->typeEvent;
    }

    public function setTypeEvent(string $typeEvent): self
    {
        $this->typeEvent = $typeEvent;
        return $this;
    }

    public function getImageEvent(): ?string
    {
        return $this->imageEvent;
    }

    public function setImageEvent(?string $imageEvent): self
    {
        $this->imageEvent = $imageEvent;
        return $this;
    }

    public function getPrixEvent(): ?string
    {
        return $this->prixEvent;
    }

    public function setPrixEvent(?string $prixEvent): self
    {
        $this->prixEvent = $prixEvent;
        return $this;
    }

    public function getCreatedAt(): \DateTimeImmutable
    {
        return $this->createdAt;
    }

    public function setCreatedAt(\DateTimeImmutable $createdAt): self
    {
        $this->createdAt = $createdAt;
        return $this;
    }
}
